### PR TITLE
Add Shadow Locale to Settings Tab

### PR DESCRIPTION
**Pull request by alexander-schranz**
To provide a shadow locale feature we add a new ShadowInterface for this which provides this feature. Then new `shadowLocale` and `shadowLocales` feature is added to the DimensionContent entities and added to the Settings tab.

In upcoming Pull Request the shadow behaviour while publishing should be implemented and checked how we handle the `url` here.

### CODE REVIEW

File: Content/Domain/Model/ShadowTrait.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+{
+    /**
+     * @var string|null
+     */
+    protected $shadowLocale;
+
+    /**
+     * @var string[]|null
+     */
+    protected $shadowLocales = null;
```
Comment (Prokyonn) [inline Content/Domain/Model/ShadowTrait.php:26]: Wouldn't it be more convenient to have the default value set to an empty array, then we could omit the check in line 46

Comment (alexander-schranz) [reply Content/Domain/Model/ShadowTrait.php:26]: @Prokyonn there should not exis default values in the dimensions. Not set values are null. This make the dimension table more readable and see which dimension properties are used in which dimension itself and make merging of the dimension easier. So `shadowLocales` will be null on the localized dimension as it should never exist there and only set on the unlocalized dimension.

Same already is done for `availableLocales`.

File: Content/Application/ContentNormalizer/Normalizer/ShadowNormalizer.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        if (!$object instanceof ShadowInterface) {
+            return $normalizedData;
+        }
+
+        Assert::isInstanceOf($object, DimensionContentInterface::class);
+
+        $normalizedData['shadowOn'] = null !== $object->getShadowLocale();
+        $normalizedData['shadowLocales'] = $normalizedData['shadowLocales'] ?? [];
+        $normalizedData['contentLocales'] = $object->getAvailableLocales() ?? []; // TODO should be changed in Sulu Core (PageSettingsShadowLocaleSelect.js)
```
Comment (alexander-schranz) [inline Content/Application/ContentNormalizer/Normalizer/ShadowNormalizer.php:32]: Issue: https://github.com/sulu/sulu/issues/7040

File: Resources/translations/admin.de.json [hunk 3]
```diff
@@ -5,6 +5,10 @@
     "sulu_content.published": "Veröffentlicht am",
     "sulu_content.author": "Autor",
     "sulu_content.authored_date": "Verfasst am",
+    "sulu_content.shadow_page": "Shadow Seite",
+    "sulu_content.enable_shadow_page": "Shadow Seite aktivieren",
+    "sulu_content.enable_shadow_page_info_text": "",
+    "sulu_content.shadow_locale": "Shadow sprache",
```
Comment (Prokyonn) [inline Resources/translations/admin.de.json]:
```suggestion
    "sulu_content.shadow_locale": "Shadow Sprache",
```

File: Resources/translations/admin.de.json [hunk 4]
```diff
@@ -5,6 +5,10 @@
     "sulu_content.published": "Veröffentlicht am",
     "sulu_content.author": "Autor",
     "sulu_content.authored_date": "Verfasst am",
+    "sulu_content.shadow_page": "Shadow Seite",
+    "sulu_content.enable_shadow_page": "Shadow Seite aktivieren",
+    "sulu_content.enable_shadow_page_info_text": "",
```
Comment (Prokyonn) [inline Resources/translations/admin.de.json]:
```suggestion
    "sulu_content.enable_shadow_page_info_text": "Diese Funktion ist deaktiviert, wenn die aktuelle Sprache bereits als Shadow verwendet wird.",
```

**Review by Prokyonn — APPROVED** [review]

**Pull request merged**